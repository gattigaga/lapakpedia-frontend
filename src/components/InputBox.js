import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Container = styled.div`
  width: 100%;
`;

export const Input = styled.input`
  border: 0px;
  border-bottom: 1px solid ${props => (props.isError ? "red" : "#ccc")};
  outline: none;
  font-size: 14px;
  letter-spacing: 0.1em;
  padding: 8px 0px;
  width: 100%;
`;

export const Helper = styled.p`
  margin: 0px;
  margin-top: 2px;
  font-family: Roboto;
  font-size: 11px;
  color: ${props => (props.isError ? "red" : "#aaa")};
`;

const InputBox = ({
  placeholder,
  type,
  value,
  helperText,
  onChange,
  isError,
  isDisabled
}) => (
  <Container>
    <Input
      type={type}
      value={value}
      onChange={onChange}
      placeholder={placeholder}
      spellcheck="false"
      isError={isError}
      disabled={isDisabled}
    />
    {helperText && <Helper isError={isError}>{helperText}</Helper>}
  </Container>
);

InputBox.propTypes = {
  placeholder: PropTypes.string,
  value: PropTypes.string,
  type: PropTypes.string,
  helperText: PropTypes.string,
  onChange: PropTypes.func,
  isError: PropTypes.bool,
  isDisabled: PropTypes.bool
};

InputBox.defaultProps = {
  placeholder: "Placeholder"
};

export default InputBox;
