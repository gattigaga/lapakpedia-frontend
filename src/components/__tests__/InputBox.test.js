import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import InputBox, { Input } from "components/InputBox";

describe("InputBox", () => {
  const setup = propOverrides => {
    const props = {
      ...propOverrides
    };

    const wrapper = shallow(<InputBox {...props} />);

    return {
      props,
      wrapper
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders with helper text", () => {
    const { wrapper } = setup({ helperText: "It's my helper text" });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders with custom placeholder", () => {
    const { wrapper } = setup({ placeholder: "Username" });

    expect(wrapper.find(Input).props().placeholder).toEqual("Username");
  });

  it("should renders with type 'password'", () => {
    const { wrapper } = setup({ type: "password" });

    expect(wrapper.find(Input).props().type).toEqual("password");
  });

  it("should renders in error state", () => {
    const { wrapper } = setup({
      isError: true,
      helperText: "It's my helper text"
    });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders ", () => {
    const { wrapper } = setup({ isDisabled: true });

    expect(wrapper.find(Input).props().disabled).toEqual(true);
  });
});
