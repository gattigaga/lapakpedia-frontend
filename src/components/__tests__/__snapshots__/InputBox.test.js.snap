// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox should renders default 1`] = `
<styled.div>
  <styled.input
    placeholder="Placeholder"
    spellcheck="false"
  />
</styled.div>
`;

exports[`InputBox should renders in error state 1`] = `
<styled.div>
  <styled.input
    isError={true}
    placeholder="Placeholder"
    spellcheck="false"
  />
  <styled.p
    isError={true}
  >
    It's my helper text
  </styled.p>
</styled.div>
`;

exports[`InputBox should renders with helper text 1`] = `
<styled.div>
  <styled.input
    placeholder="Placeholder"
    spellcheck="false"
  />
  <styled.p>
    It's my helper text
  </styled.p>
</styled.div>
`;
